File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/kundenstamm/kunden/dtos/i-kunde.ts
import { ApiKunde } from './api/api-kunde';

export interface IKunde {
    id: string;
    name?: string;
    boolean?: boolean;
    dateTime?: Date;
    double?: number;
    guid?: string;
    integer?: number;
    bankId: string;
}

export class Kunde {
    public static fromApiKunde(apiKunde: ApiKunde): IKunde {
        return {
            id: apiKunde.id,
            name: apiKunde.name,
            boolean: apiKunde.boolean,
            dateTime: apiKunde.dateTime,
            double: apiKunde.double,
            guid: apiKunde.guid,
            integer: apiKunde.integer,
            bankId: apiKunde.bankId,
        };
    }
}

File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/Kundenstamm/Kunden/DTOs/IKunde.cs
using System;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.Kundenstamm.Kunden
{
    public interface IKunde
    {
        Guid Id { get; set; }

        string? Name { get; set; }

        bool? Boolean { get; set; }

        DateTime? DateTime { get; set; }

        double? Double { get; set; }

        Guid? Guid { get; set; }

        int? Integer { get; set; }

        Guid BankId { get; set; }
    }
}